### PR TITLE
Domains: Inbound Transfer Edit screen

### PR DESCRIPTION
**Pull request by aidvu**
Add a screen for failed transfer that allows users to
restart the transfer and contact support.

Update the domain warning.

failed-edit-screen

Test:
1. Start a domain transfer
2. Fail it on the backend
3. Try restarting
4. Check messaging
5. Try to properly restart the transfer.

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 1]
```diff
@@ -43,26 +44,76 @@ class Transfer extends React.PureComponent {
...
+									a: <a href={ '#' } />,
+								},
+								args: { domain: domain.name },
+							}
+						) }
+					</p>
+				</div>
+				<div>
+					<Button onClick={ this.restartTransfer }>
+						{ this.props.translate( 'Start Transfer Again' ) }
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Hi! I've found a possible matching string that has already been translated **11** times:
`translate( 'Start my Guided Transfer' )` *ES Score: 7*

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 2]
```diff
@@ -43,26 +44,76 @@ class Transfer extends React.PureComponent {
...
+		} );
+	};
+
+	getCancelledContent() {
+		const { domain, translate } = this.props;
+
+		return (
+			<Card>
+				<div>
+					<p className="edit__transfer_text_fail">{ translate( 'Domain Transfer Failed' ) }</p>
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Domain transfer failed' )` *ES Score: 17*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/failed-transfer-screen)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

Comment (hambai) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: You should change the `className` to `edit__transfer-text-fail` - the underscores are the reason CI fails

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 3]
```diff
@@ -43,26 +44,76 @@ class Transfer extends React.PureComponent {
...
+								args: { domain: domain.name },
+							}
+						) }
+					</p>
+				</div>
+				<div>
+					<Button onClick={ this.restartTransfer }>
+						{ this.props.translate( 'Start Transfer Again' ) }
+					</Button>
+					<Button className="edit__transfer_button_margin" href={ support.CALYPSO_CONTACT }>
```
Comment (hambai) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: And here too - `edit__transfer-button-margin`

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 4]
```diff
@@ -43,26 +44,76 @@ class Transfer extends React.PureComponent {
...
+		const { domain, translate } = this.props;
+
+		return (
+			<Card>
+				<div>
+					<p className="edit__transfer_text_fail">{ translate( 'Domain Transfer Failed' ) }</p>
+					<p>
+						{ translate(
+							'We were unable to verify the transfer of {{strong}}%(domain)s{{/strong}}. The domain ' +
+								'authorization code from you current registrar was not provided to initiate the transfer. ' +
```
Comment (hambai) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: I wonder if we should put something like "or the transfer was cancelled" here?

File: client/my-sites/domains/domain-management/edit/transfer.jsx (hunk 2 above)
Comment (aidvu) [reply client/my-sites/domains/domain-management/edit/transfer.jsx]: :man_facepalming:

File: client/my-sites/domains/domain-management/edit/transfer.jsx (hunk 4 above)
Comment (aidvu) [reply client/my-sites/domains/domain-management/edit/transfer.jsx]: @danhauk good point for copy here.

File: client/my-sites/domains/domain-management/edit/transfer.jsx [hunk 5]
```diff
@@ -43,26 +44,76 @@ class Transfer extends React.PureComponent {
...
+		} );
+	};
+
+	getCancelledContent() {
+		const { domain, translate } = this.props;
+
+		return (
+			<Card>
+				<div>
+					<p className="edit__transfer-text-fail">{ translate( 'Domain Transfer Failed' ) }</p>
```
Comment (a8ci18n) [inline client/my-sites/domains/domain-management/edit/transfer.jsx]: Hi! I've found a possible matching string that has already been translated **17** times:
`translate( 'Domain transfer failed' )` *ES Score: 17*
See 1 additional suggestion in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/update/failed-transfer-screen)

Help me improve these suggestions: react with :thumbsdown: if the suggestion doesn't make any sense, or with :thumbsup: if it's a particularly good one (even if not implemented).

File: client/my-sites/domains/domain-management/edit/transfer.jsx (hunk 4 above)
Comment (danhauk) [reply client/my-sites/domains/domain-management/edit/transfer.jsx]: @aidvu Ideally if we actually _know_ the reason it failed we customize the message based on that. If we can't get that, then adding @hambai's suggestion is a good compromise.

Comment (aidvu) [reply client/my-sites/domains/domain-management/edit/transfer.jsx]: @danhauk we should make a general message for now, and see what kind of errors we get, and if they are any good for specific messaging.

Comment (danhauk) [reply client/my-sites/domains/domain-management/edit/transfer.jsx]: @aidvu What other kinds of reasons could we have for the transfer failing? Would adding "or the transfer was cancelled" be enough or could there be other reasons?

Comment (aidvu) [reply client/my-sites/domains/domain-management/edit/transfer.jsx]: Hmm. Domain relocked. Time passed and was never confirmed (so not really the same as cancelled).

**Pull request merged**